csharp/src/Ice/WSEndpoint.cs

```diff
@@ -172,140 +111,128 @@ public override IAcceptor Acceptor(string adapterName)
             return new WSAcceptor(this, _instance, acceptor);
         }
 
-        public WSEndpoint endpoint(Endpoint delEndp)
+        public override void ConnectorsAsync(Ice.EndpointSelectionType endpointSelection, IEndpointConnectors callback)
         {
-            if (delEndp == _delegate)
-            {
-                return this;
-            }
-            else
+            string host = "";
+            for (Ice.Endpoint? p = _delegate; p != null; p = p.Underlying)
             {
-                return new WSEndpoint(_instance, delEndp, _resource);
+                if (p is Ice.IPEndpoint ipEndpoint)
+                {
+                    host = $"{ipEndpoint.Host}:{ipEndpoint.Port.ToString(CultureInfo.InvariantCulture)}";
+                    break;
+                }
             }
+            _delegate.ConnectorsAsync(endpointSelection, new EndpointConnectors(_instance, host, Resource, callback));
         }
 
+        public WSEndpoint GetEndpoint(Endpoint del)
+            => del == _delegate ? this : new WSEndpoint(_instance, del, Resource);
```

Comment:
```suggestion
        public WSEndpoint GetEndpoint(Endpoint del) => 
            del == _delegate ? this : new WSEndpoint(_instance, del, Resource);
```

Don't we put the `=>` arrow in the above line always?

Reply:
Yes, you are right. See for example: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Threading.Channels/src/System/Threading/Channels/Channel.cs